dynamiqs/utils/utils.py

```diff
@@ -412,17 +414,60 @@ def expect(O: Tensor, x: Tensor) -> Tensor:
         is real. One can then keep only the real values of the returned tensor using
         `dq.expect(O, x).real`.
 
-    Warning:
-        This function does not yet support tensors in the bra format.
-
     Args:
         O (n, n): Arbitrary operator.
-        x (..., n, 1) or (..., n, n): Ket or density matrix.
+        x (..., n, 1) or (..., 1, n) or (..., n, n): Ket, bra or density matrix.
 
     Returns:
         (...): Complex-valued expectation value.
+
+    Raises:
+        TypeError: If the input tensor is not a ket, bra or density matrix.
     """
-    # TODO: adapt to bras
     if is_ket(x):
         return torch.einsum('...ij,jk,...kl->...', x.mH, O, x)  # <x|O|x>
-    return torch.einsum('ij,...ji->...', O, x)  # tr(Ox)
+    elif is_bra(x):
+        return torch.einsum('...ij,jk,...kl->...', x, O, x.mH)
+    elif is_dm(x):
+        return torch.einsum('ij,...ji->...', O, x)  # tr(Ox)
+    else:
+        raise TypeError('Input tensor is not a ket, bra or density matrix.')
+
+
+def norm(x: Tensor) -> Tensor:
+    r"""Returns the norm of a ket, bra or a density matrix.
+
+    For kets and bras, the returned norm is $\sqrt{\braket{\psi|\psi}}$. For density
+    matrices, it is $\mathrm{Tr} (\rho)$.
```

Comment:
```suggestion
    matrices, it is $\tr{\rho}$.
```
See shortcuts defined here > https://github.com/dynamiqs/dynamiqs/blob/main/docs/javascripts/mathjax.js#L10.